### PR TITLE
Fix intermittent failure in wallet_send.py and rpc_fundrawtransaction.py

### PR DESCRIPTION
**Pull request by meshcollider**
After #17211 was merged, there have been a few intermittent test failures in `wallet_send.py`, `rpc_fundrawtransaction.py`, and `rpc_psbt.py`

I believe all these failures are due to these missing `sync_all()`s

### CODE REVIEW

**Comment by achow101** [comment]
`rpc_psbt.py` has a similar potential issue.

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -1011,6 +1011,7 @@ def test_external_inputs(self):
         self.nodes[0].sendtoaddress(addr, 10)
         self.nodes[0].sendtoaddress(wallet.getnewaddress(), 10)
         self.nodes[0].generate(6)
```
[the remark this answers is not in the document]
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]:
```
$ git grep "self.nodes\[0\]\.generate"
test/functional/interface_zmq.py:586:        self.nodes[0].generatetoaddress(1, ADDRESS_BCRT1_UNSPENDABLE)
test/functional/rpc_fundrawtransaction.py:1013:        self.nodes[0].generate(6)
test/functional/rpc_psbt.py:630:        self.nodes[0].generate(6)
test/functional/wallet_send.py:505:        self.nodes[0].generate(6)
```

**Comment by meshcollider** [comment]
Added `rpc_psbt.py` and addressed @jonatack's suggestion

**Comment by meshcollider** [comment]
Sorry to invalidate your ACK immediately @achow101, just included the two other instances of node.generate that @jonatack pointed out so they're all gone.

**Comment by achow101** [comment]
ACK 75a9305d455e234c6b63635d80b0f2aef902342e

**Comment by instagibbs** [comment]
the idea being block processing hasn't finished before the wallet calls are used, even though it's on the same node?

**Comment by meshcollider** [comment]
> the idea being block processing hasn't finished before the wallet calls are used, even though it's on the same node?

It is for the benefit of the other node, using the input externally. For example, in `rpc_fundrawtransaction.py`, the node called `wallet` is the one calling `fundrawtransaction`, after node[0] generates the transaction and blocks

File: test/functional/rpc_fundrawtransaction.py [hunk 2]
```diff
@@ -1010,7 +1010,8 @@ def test_external_inputs(self):
 
         self.nodes[0].sendtoaddress(addr, 10)
         self.nodes[0].sendtoaddress(wallet.getnewaddress(), 10)
-        self.nodes[0].generate(6)
+        self.generate(self.nodes[0], 6)
+        self.sync_all()
```
[the remark this answers is not in the document]
Comment (meshcollider) [reply test/functional/rpc_fundrawtransaction.py:1014]: @instagibbs  #22567 will merge these `sync_all` into the `generate` calls, so it makes sense to put them together for clarity at the moment IMO

**Comment by instagibbs** [comment]
Got it.

**Review by lsilva01 — APPROVED** [review]
Tested ACK https://github.com/bitcoin/bitcoin/pull/23200/commits/75a9305d455e234c6b63635d80b0f2aef902342e on Ubuntu 20.04

`sync_all()` ensures that blocks and mempool will be synchronized between nodes.
Tests were run several times without errors.

File: test/functional/rpc_fundrawtransaction.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lsilva01) [reply test/functional/rpc_fundrawtransaction.py:1014]: @instagibbs  PR https://github.com/bitcoin/bitcoin/pull/22741 has more information about the `generate*` calls

**Pull request merged**